prow/deck/jobs/jobs.go

```diff
@@ -294,8 +299,22 @@ func (ja *JobAgent) update() error {
 			duration -= duration % time.Second // strip fractional seconds
 			nj.Duration = duration.String()
 		}
+		if j.ObjectMeta.Annotations[client.GerritID] != "" &&
+			j.ObjectMeta.Annotations[client.GerritInstance] != "" &&
+			j.ObjectMeta.Labels[client.GerritRevision] != "" {
+			nj.HostType = "gerrit"
+			nj.ReviewHost = j.ObjectMeta.Annotations[client.GerritInstance]
+			parts := strings.SplitN(nj.ReviewHost, ".", 2)
+			nj.CodeHost = strings.TrimSuffix(parts[0], "-review")
+			if len(parts) > 1 {
+				nj.CodeHost += "." + parts[1]
+			}
```

Comment:
add a unit-test for the expected fields?

Reply:
done